From 1e322b883e036fe0bd5468fe60a0431545fe2376 Mon Sep 17 00:00:00 2001
From: Marc Dionne <marc.c.dionne@gmail.com>
Date: Sun, 27 Mar 2011 11:20:17 -0400
Subject: [PATCH 1/4] Linux: 2.6.39: deal with BKL removal

For 2.6 kernels the OpenAFS code is already BKL free, but the
corresponding header file smp_lock.h is still included in various
places.  This header no longer exists, so remove all occurrences
outside of LINUX24.

Reviewed-on: http://gerrit.openafs.org/4361
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Derrick Brashear <shadow@dementia.org>
(cherry picked from commit 9376293634393597ac7db88593100937621f6c3f)

Change-Id: I8e7e244b2e45a7a3af0505531757b8a54a0b3d29
Reviewed-on: http://gerrit.openafs.org/4370
Tested-by: BuildBot <buildbot@rampaginggeek.com>
Reviewed-by: Derrick Brashear <shadow@dementia.org>
---
 src/afs/LINUX/osi_export.c   |    1 -
 src/afs/LINUX/osi_file.c     |    1 -
 src/afs/LINUX/osi_groups.c   |    2 --
 src/afs/LINUX/osi_misc.c     |    1 -
 src/afs/LINUX/osi_nfssrv.c   |    1 -
 src/afs/LINUX/osi_vfsops.c   |    1 -
 src/afs/LINUX/osi_vnodeops.c |    1 -
 src/afs/afs_call.c           |    3 ---
 src/afs/afs_icl.c            |    3 ---
 src/afs/afs_pag_call.c       |    3 ---
 src/afs/afs_syscall.c        |    3 ---
 src/rx/LINUX/rx_knet.c       |    1 -
 12 files changed, 0 insertions(+), 21 deletions(-)

diff --git a/src/afs/LINUX/osi_export.c b/src/afs/LINUX/osi_export.c
index 5b300e0..662966b 100644
--- a/src/afs/LINUX/osi_export.c
+++ b/src/afs/LINUX/osi_export.c
@@ -23,7 +23,6 @@
 #include "afs/sysincludes.h"
 #include "afsincludes.h"
 #include "afs/afs_dynroot.h"
-#include <linux/smp_lock.h>
 
 #if !defined(AFS_NONFSTRANS)
 
diff --git a/src/afs/LINUX/osi_file.c b/src/afs/LINUX/osi_file.c
index 4a28da6..fbd20e9 100644
--- a/src/afs/LINUX/osi_file.c
+++ b/src/afs/LINUX/osi_file.c
@@ -15,7 +15,6 @@
 #include "afs/sysincludes.h"	/* Standard vendor system headers */
 #include "afsincludes.h"	/* Afs-based standard headers */
 #include "afs/afs_stats.h"	/* afs statistics */
-#include <linux/smp_lock.h>
 #include <linux/namei.h>
 
 #if defined(HAVE_LINUX_EXPORTFS_H)
diff --git a/src/afs/LINUX/osi_groups.c b/src/afs/LINUX/osi_groups.c
index 7561ecc..144f0e3 100644
--- a/src/afs/LINUX/osi_groups.c
+++ b/src/afs/LINUX/osi_groups.c
@@ -26,8 +26,6 @@
 #include "afs/nfsclient.h"
 #include "osi_compat.h"
 
-#include <linux/smp_lock.h>
-
 #ifdef AFS_LINUX26_ONEGROUP_ENV
 # define NUMPAGGROUPS 1
 
diff --git a/src/afs/LINUX/osi_misc.c b/src/afs/LINUX/osi_misc.c
index e2af595..79fa763 100644
--- a/src/afs/LINUX/osi_misc.c
+++ b/src/afs/LINUX/osi_misc.c
@@ -22,7 +22,6 @@
 #include "afs/sysincludes.h"
 #include "afsincludes.h"
 #include "afs/afs_stats.h"
-#include <linux/smp_lock.h>
 
 int afs_osicred_initialized = 0;
 afs_ucred_t afs_osi_cred;
diff --git a/src/afs/LINUX/osi_nfssrv.c b/src/afs/LINUX/osi_nfssrv.c
index 4c0553b..cde4b9b 100644
--- a/src/afs/LINUX/osi_nfssrv.c
+++ b/src/afs/LINUX/osi_nfssrv.c
@@ -21,7 +21,6 @@
 #include "afs/sysincludes.h"
 #include "afsincludes.h"
 #include "nfsclient.h"
-#include <linux/smp_lock.h>
 #include <linux/sunrpc/svc.h>
 #include <linux/sunrpc/svcauth.h>
 
diff --git a/src/afs/LINUX/osi_vfsops.c b/src/afs/LINUX/osi_vfsops.c
index 921efac..8abe6b1 100644
--- a/src/afs/LINUX/osi_vfsops.c
+++ b/src/afs/LINUX/osi_vfsops.c
@@ -21,7 +21,6 @@
 #include "afs/sysincludes.h"
 #include "afsincludes.h"
 #include "afs/afs_stats.h"
-#include <linux/smp_lock.h>
 
 #include "osi_compat.h"
 
diff --git a/src/afs/LINUX/osi_vnodeops.c b/src/afs/LINUX/osi_vnodeops.c
index f3ba62a..d8d23b5 100644
--- a/src/afs/LINUX/osi_vnodeops.c
+++ b/src/afs/LINUX/osi_vnodeops.c
@@ -30,7 +30,6 @@
 #include <linux/mm_inline.h>
 #endif
 #include <linux/pagemap.h>
-#include <linux/smp_lock.h>
 #include <linux/writeback.h>
 #include <linux/pagevec.h>
 #include "afs/lock.h"
diff --git a/src/afs/afs_call.c b/src/afs/afs_call.c
index 5c1538e..f2778fe 100644
--- a/src/afs/afs_call.c
+++ b/src/afs/afs_call.c
@@ -24,9 +24,6 @@
 #include "netinet/in_var.h"
 #endif
 #endif /* !defined(UKERNEL) */
-#ifdef AFS_LINUX22_ENV
-#include "h/smp_lock.h"
-#endif
 #ifdef AFS_SUN510_ENV
 #include "h/ksynch.h"
 #include "h/sunddi.h"
diff --git a/src/afs/afs_icl.c b/src/afs/afs_icl.c
index c8547ef..7c3d175 100644
--- a/src/afs/afs_icl.c
+++ b/src/afs/afs_icl.c
@@ -24,9 +24,6 @@
 #include "netinet/in_var.h"
 #endif
 #endif /* !defined(UKERNEL) */
-#ifdef AFS_LINUX22_ENV
-#include "h/smp_lock.h"
-#endif
 
 
 struct afs_icl_set *afs_iclSetp = (struct afs_icl_set *)0;
diff --git a/src/afs/afs_pag_call.c b/src/afs/afs_pag_call.c
index b1ea882..d61a91c 100644
--- a/src/afs/afs_pag_call.c
+++ b/src/afs/afs_pag_call.c
@@ -25,9 +25,6 @@
 #include "netinet/in_var.h"
 #endif
 #endif /* !defined(UKERNEL) */
-#ifdef AFS_LINUX22_ENV
-#include "h/smp_lock.h"
-#endif
 #include "rmtsys.h"
 #include "pagcb.h"
 
diff --git a/src/afs/afs_syscall.c b/src/afs/afs_syscall.c
index b6f6337..8ad59e9 100644
--- a/src/afs/afs_syscall.c
+++ b/src/afs/afs_syscall.c
@@ -28,9 +28,6 @@
 #include "netinet/in_var.h"
 #endif
 #endif /* !defined(UKERNEL) */
-#ifdef AFS_LINUX22_ENV
-#include "h/smp_lock.h"
-#endif
 
 #if (defined(AFS_AIX51_ENV) && defined(AFS_64BIT_KERNEL)) || defined(AFS_HPUX_64BIT_ENV) || defined(AFS_SUN57_64BIT_ENV) || (defined(AFS_SGI_ENV) && (_MIPS_SZLONG==64)) || defined(NEED_IOCTL32)
 static void
diff --git a/src/rx/LINUX/rx_knet.c b/src/rx/LINUX/rx_knet.c
index 590a060..942a072 100644
--- a/src/rx/LINUX/rx_knet.c
+++ b/src/rx/LINUX/rx_knet.c
@@ -18,7 +18,6 @@
 
 #include <linux/version.h>
 #include "rx/rx_kcommon.h"
-#include "h/smp_lock.h"
 #include <asm/uaccess.h>
 #ifdef ADAPT_PMTU
 #include <linux/errqueue.h>
-- 
1.7.4.1

